internal/cluster/cluster.go

```diff
@@ -316,11 +317,26 @@ func (m *Manager) RefreshStatus(ctx context.Context, c *common.Cluster, db *gorm
 	if db == nil {
 		db = m.db
 	}
-	vc, err := newClusterValidationContext(*c.ID, db)
+	cluster, err := common.GetClusterFromDBWithoutDisabledHosts(db, *c.ID)
 	if err != nil {
-		return c, err
+		return nil, err
 	}
-	conditions, newValidationRes, err := m.rp.preprocess(ctx, vc)
+	return m.refreshStatusInternal(ctx, cluster, db)
+}
+
+func (m *Manager) refreshStatusInternal(ctx context.Context, c *common.Cluster, db *gorm.DB) (*common.Cluster, error) {
+	//new transition code
+	if db == nil {
+		db = m.db
+	}
+	var (
+		vc               *clusterPreprocessContext
+		err              error
+		conditions       map[string]bool
+		newValidationRes map[string][]ValidationResult
+	)
+	vc = newClusterValidationContext(c, db)
```

Comment:
This function loads the cluster from the DB as well. maybe it should get it as a parameter?